sympy/ntheory/generate.py

```diff
@@ -219,10 +242,31 @@ def prime(nth):
         primerange : Generate all primes in a given range
         primepi : Return the number of primes less than or equal to n
     """
-    n = as_int(nth)
+    n = int(nth)
     if n < 1:
         raise ValueError("nth must be a positive integer; prime(1) == 2")
-    return sieve[n]
+    if n<=2:
+        return n+1
+    from sympy.functions.special.error_functions import li
```

Comment:
Imports on the top was creating problems.
`from sympy import sieve` wasn't working after that.